shared/styles/index.js.flow

```diff
@@ -48,7 +48,16 @@ declare export var statusBarHeight: number
 declare export var hairlineWidth: number
 declare export function backgroundURL(...path: Array<string>): string
 declare export function styleSheetCreate(obj: Object): Object
-declare export function collapseStyles(styles: any): any
+
+type _Elem = Object | null | false | void
+// CollapsableStyle is a generic version of ?StylesMobile and family,
+// slightly extended to support "isFoo && myStyle".
+export type CollapsibleStyle = _Elem | $ReadOnlyArray<_Elem>
+// This has to return any, since there are still components with style
+// fields erroneously typed to not accept an array.
+//
+// TODO: Fix all of those, then type the return value.
+declare export function collapseStyles(styles: $ReadOnlyArray<CollapsibleStyle>): any
```

Comment:
Currently, we always pass in an array to `collapseStyles`, even though it was typed in desktop/mobile to take in a single object. But there's no reason to call `collapseStyles` on a single object, so just make it take an array.